### ISSUE TITLE
Pod doesn't look directory outside

### ISSUE THREAD

**Issue by caatclaudia** [operator-framework/operator-sdk#5828]
## Context
I'm creating a Kubernetes operator, through Ansible, that allows me to monitor the node's state from a Kubernetes Cluster, namely that allows me to change files in a directory from the node Cluster (not inside the pod). For this, my main.yml file executes the following commands:

```
- name: Change Nodes Pods
  lineinfile:
    path: ~/Downloads/files/*
    line: "  nodeName: node01" 
```

When I use these commands, the pod cannot execute the commands correctly, since the pod looks for this directory inside and not on the machine where the pod runs.
```
make docker-build docker-push IMG=<DOCKER USER>/<REPOSITORY>:latest
make deploy IMG=<DOCKER USER>/<REPOSITORY>:latest
kubectl get deployment -n <NAMESPACE>
kubectl create -f config/samples/<FILE NAME>.yaml -n <NAMESPACE>
kubectl get pods -n <NAMESPACE>
kubectl logs --follow <POD ID> -n <NAMESPACE>
```

## Question
How can I resolve this issue? How can I set the pod to go to the outside board? Or how can I create a volume for file sharing?

**Comment by everettraven** [comment]
@caatclaudia I am not aware of any ways for a pod to reach the host filesystem other than potentially mounting a volume. I think the way to do this would be to modify the `config/manager/manager.yaml` file which contains the `Deployment` for your operator. You should be able to modify this `Deployment` definition to mount whatever volumes your project may need.

An example of this file can be found in the Ansible testdata for Operator SDK here: https://github.com/operator-framework/operator-sdk/blob/master/testdata/ansible/memcached-operator/config/manager/manager.yaml

Here is where I believe any modifications should go: https://github.com/operator-framework/operator-sdk/blob/8c9f24ca0f75b05ad5101d711f69253375864be7/testdata/ansible/memcached-operator/config/manager/manager.yaml#L52-L53

Let me know if this helps and does what you are looking for!

**Comment by caatclaudia** [comment]
Thank you, this helps me a lot. But the only problem is that I want to make some changes in files, and the volume path has the root owner, but my operator runs in ansible user, so it doesn't have permissions to write in files. Any suggestion?

**Comment by everettraven** [comment]
@caatclaudia Hmm... I am not entirely sure on this one. It may be that you need to change the permissions on the files but I will look into this more and get back to you.

**Comment by caatclaudia** [comment]
I already have found the answer, so I'm gonna close this issue, but I am gonna put the answer here to maybe help someone that can have the same issues.

I have seen this [link](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ ) that is about **security context**, this security context allows the definition of the user and group that are running this container. I found some solutions that said if I put this Security context before the definition of the volumes, the volumes will be created with the user and group that are defined, but I tried and this doesn't happen, my volumes have the root owner anyway.

So, my solution (which works but is not perfect in terms of security) was to run the container with the root user, to have permissions on my volumes. Like this:
```
spec:
  securityContext:
    runAsUser: 0
    runAsGroup: 0
    fsGroup: 0
  volumes:
  ....
```
